command/format/diff_test.go

```diff
@@ -935,6 +1273,118 @@ func TestResourceChange_nestedSet(t *testing.T) {
             volume_type = "gp2"
         }
     }
+`,
+		},
+		"force-new update (inside block)": {
+			Action: plans.DeleteThenCreate,
+			Mode:   addrs.ManagedResourceMode,
+			Before: cty.ObjectVal(map[string]cty.Value{
+				"id":  cty.StringVal("i-02ae66f368e8518a9"),
+				"ami": cty.StringVal("ami-BEFORE"),
+				"root_block_device": cty.SetVal([]cty.Value{
+					cty.ObjectVal(map[string]cty.Value{
+						"volume_type": cty.StringVal("gp2"),
+					}),
+				}),
+			}),
+			After: cty.ObjectVal(map[string]cty.Value{
+				"id":  cty.StringVal("i-02ae66f368e8518a9"),
+				"ami": cty.StringVal("ami-AFTER"),
+				"root_block_device": cty.SetVal([]cty.Value{
+					cty.ObjectVal(map[string]cty.Value{
+						"volume_type": cty.StringVal("different"),
+					}),
+				}),
+			}),
+			RequiredReplace: cty.NewPathSet(cty.Path{
+				cty.GetAttrStep{Name: "root_block_device"},
+				cty.IndexStep{Key: cty.NumberIntVal(0)},
```

Comment:
Since `root_block_device` is defined as a set, this is not a realistic path for it... there isn't any way to draw a path through a set (since they don't have keys) and so in "real life" the set as a whole is what would be here, with all of the separate block adds/removes marked as "forces replacement" in the output then.

This test _does_ make sense for a block of `NestingList`, though.